protected_app_signals_cost.md

```diff
@@ -0,0 +1,168 @@
+### F: Extra cost of running the Buyer RTB Server
+
+There will be changes both in the RTB processing of the PAS request and in the size of the data to be set to the RTB server. We assume this difference in cost to be negligible.
+
+### L: Cost of running the BFE
+
+This is the [VM Cost][12] marked C2 and C5 in the B&A explainer. With PAS, the BFE is running an additional flow to PA, thus incurring extra cost. The cost of the PAS flow could be considered equivalent to the cost of processing an extra CustomAudience (or Interest Group in the Chrome parlance).
+
+### N: Cost of running the Bidding Server
+
+This is where we expect noticeable differences, mostly because of the computation cost. Initial estimates suggest the AppInstall bidding server may require ~70% additional computational power for multiple javascript executions. We should account the extra workload required to run more complex inference
+
+The distribution of the cost for running the Bidding Server based on internal estimates is shown below:
+
+The three factors are:
+
+*   [Inference][15]. This is expected to constiture the vast majority of the cost. Details on how to compute the inference costs will be added later.
+*   Cost of preparing the data for ad retrieval: essentially the cost for running the. prepareDataForAdRetrievalUDF. This is a minor contributor.
+*   Base cost for running the bidding process, this is the cost for coordinating the flow, processing the data from the ads retrieval server and sending it to generateBid, handling the response and returning the results to the BFE.
+
+There are a lot of changes in bidding server
+
+1.  prepareDataForAdRetrieval - This is a new function that runs before generateBid when using TEE based retrieval and it is skipped in case of contextual based retrieval. The function is intended to be used to decode the signals received from the device and make them available to the retrieval server UDF and to the generateBid. We are expecting that this function might run some inference too. The main factor to consider when estimating this cost are:
+    1.  Complexity of the model used for feature extraction
+    2.  Amount of signals received (will likely always be the currently max allowed size of 1.5KB but we are working on enabling the seller to control the payload size alloted for each buyer so this might change).
+    3.  Logic and language used to implement the UDF (using JS or WASM)
+    4.  Cost of communicating with the inference service (sidecar process at the moment)
+2.  Sending request to retrieval server (marked as O in the diagram above) - cost is zero
+3.  Getting info back from retrieval and sending it to generateBid. This may involve data copying into v8. We need to figure out how much data this is, and what the copying costs are (response size from retrieval, size of embeddings, number of embeddings, number of ads).
+4.  generateBid:
+    1.  Cost of the script itself - maybe assume this is similar to the remarketing case
```

Comment:
It is not clear what `Cost of script` mean. Is it the cost associated with running the script?